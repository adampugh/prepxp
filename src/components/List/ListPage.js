import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";

import * as actions from "../../store/actions/actions";
import Navbar from "../UI/navbarLoggedIn";
import ListPageHeader from "./listPageHeader";
import ListPageAddQuestionBanner from "./listPageAddQuestionBanner";
import ListPageQuestions from "./listPageQuestions";
import Footer from "../UI/footer";

export class ListPage extends Component {

    componentDidMount() {
        window.scrollTo(0, 0);
    }

    render() {
        return (
            <div>
            {
                this.props.list !== null 
                    ? <div> 
                        <Navbar />
                        <ListPageAddQuestionBanner 
                            startAddQuestion={this.props.startAddQuestion}
                            id={this.props.list.id} />
                        <ListPageHeader 
                            title={this.props.list.title} 
                            startEditListTitle={this.props.startEditListTitle}
                            id={this.props.list.id} />
                        <ListPageQuestions 
                            questions={this.props.list.questions} 
                            // questions={this.props.lists[this.props.location.state.listIndex].questions}
                            startDeleteQuestion={this.props.startDeleteQuestion} 
                            id={this.props.list.id} />
                        <Footer />
                        </div>
                    : <Redirect to="/dashboard" />
            }
                
            </div>
        )
    }
}


// <ListPageQuestions questions={this.state.questions} />

const mapStateToProps = (state, ownProps) => {
    return {
        list: ownProps.location.state ? state.lists.filter(list => list.id === ownProps.location.state.list.id)[0] : null
    }
}

const mapDispatchToProps = dispatch => {
    return {
        startAddQuestion: (id, question) => dispatch(actions.startAddQuestion(id, question)),
        startDeleteQuestion: (id, index, questionId) => dispatch(actions.startDeleteQuestion(id, index, questionId)),
        startEditListTitle: (id, title) => dispatch(actions.startEditListTitle(id, title))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ListPage);





