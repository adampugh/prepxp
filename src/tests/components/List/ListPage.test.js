import React from "react";
import { shallow } from "enzyme";

import ListPage from "../../../components/List/ListPage";

describe("<ListPage />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<ListPage />);
        expect(wrapper).toMatchSnapshot();
    });
});