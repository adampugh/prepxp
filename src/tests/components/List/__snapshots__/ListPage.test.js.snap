// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListPage /> should match snapshot 1`] = `
<div>
  <div>
    <Connect(Navbar) />
    <listPageAddQuestionBanner
      id="234"
    />
    <listPageHeader
      id="234"
      title="JavaScript Developer"
    />
    <listPageQuestions
      id="234"
      questions={
        Array [
          Object {
            "answer": "JavaScript is a language for the web.",
            "question": "What is the difference between let and var?",
          },
          Object {
            "answer": "Brendan Eich",
            "question": "Who created JavaScript?",
          },
        ]
      }
    />
    <footer />
  </div>
</div>
`;
